ports/esp32/machine_timer.c

```diff
@@ -130,20 +133,49 @@ STATIC void machine_timer_enable(machine_timer_obj_t *self) {
 
 STATIC mp_obj_t machine_timer_init_helper(machine_timer_obj_t *self, mp_uint_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
     static const mp_arg_t allowed_args[] = {
```

Comment:
To make the code clearer to read, you may want to add an enum for the args, eg: `enum { ARG_mode, ARG_callback, ARG_period, ARG_freq, ARG_period_us };`, then use them as `args[ARG_freq].u_obj` (this is done elsewhere but it's not consistently used).

Reply:
Good point. I'll do that once we finalise the API.